AcademyResidentInformationApi/V1/Controllers/TaxPayersController.cs

```diff
@@ -13,16 +13,26 @@ namespace AcademyResidentInformationApi.V1.Controllers
     public class TaxPayersController : BaseController
     {
         private readonly IGetTaxPayerByIdUseCase _getTaxPayerByIdUseCase;
+        private readonly IGetAllTaxPayersUseCase _getAllTaxPayersUseCase;
 
-        public TaxPayersController(IGetTaxPayerByIdUseCase getTaxPayerByIdUseCase)
+        public TaxPayersController(IGetAllTaxPayersUseCase getAllTaxPayersUseCase, IGetTaxPayerByIdUseCase getTaxPayerByIdUseCase)
         {
+            _getAllTaxPayersUseCase = getAllTaxPayersUseCase;
             _getTaxPayerByIdUseCase = getTaxPayerByIdUseCase;
         }
 
         [HttpGet]
-        public IActionResult ListTaxPayers()
+        public IActionResult ListTaxPayers([FromQuery] QueryParam qp)
         {
-            return Ok();
+            try
+            {
+                var record = _getAllTaxPayersUseCase.Execute(qp);
+                return Ok(record);
+            }
+            catch (TaxPayerNotFoundException)
+            {
+                return NotFound();
+            }
```

Comment:
Is this needed? When will it be thrown?

Reply:
It will be thrown in the event of a 404 error where valid parameters are given, but no tax payers are found.

Reply:
In this case it should just return an empty list

Reply:
There is also no code in this PR that actually throws that exception.

Reply:
Fair point, I think that may be part of the next ticket, good spot!